From 9669e76c2d39e5ddb6ba7af9b97599906d11eedd Mon Sep 17 00:00:00 2001
From: Adeel Arshad <embeddi.zone@gmail.com>
Date: Sun, 29 Apr 2018 08:33:20 +0100
Subject: [PATCH] invert x and y axis of usb touch panel

Signed-off-by: Adeel Arshad <embeddi.zone@gmail.com>
---
 drivers/input/touchscreen/usbtouchscreen.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/input/touchscreen/usbtouchscreen.c b/drivers/input/touchscreen/usbtouchscreen.c
index 2c41107..6a481dd 100644
--- a/drivers/input/touchscreen/usbtouchscreen.c
+++ b/drivers/input/touchscreen/usbtouchscreen.c
@@ -343,8 +343,13 @@ static int egalax_init(struct usbtouch_usb *usbtouch)
 	return ret;
 }
 
+#define INVERT_X
+#define INVERT_Y
+
 static int egalax_read_data(struct usbtouch_usb *dev, unsigned char *pkt)
 {
+	struct usbtouch_device_info *device_info = dev->type;
+
 	if ((pkt[0] & EGALAX_PKT_TYPE_MASK) != EGALAX_PKT_TYPE_REPT)
 		return 0;
 
@@ -352,6 +357,14 @@ static int egalax_read_data(struct usbtouch_usb *dev, unsigned char *pkt)
 	dev->y = ((pkt[1] & 0x0F) << 7) | (pkt[2] & 0x7F);
 	dev->touch = pkt[0] & 0x01;
 
+#ifdef INVERT_X
+	dev->x = device_info->max_xc - dev->x;
+#endif
+
+#ifdef INVERT_Y
+	dev->y = device_info->max_yc - dev->y;
+#endif
+
 	return 1;
 }
 
-- 
2.7.4

